fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/scheduler/SimpleJobScheduler.java

```diff
@@ -27,6 +32,12 @@ public class SimpleJobScheduler implements JobScheduler {
     public void executeJobPool(ScheduleContext scheduleContext) {
         JobPool pool = scheduleContext.getJobPool();
         while (!pool.isEmpty()) {
+            CascadesContext context = (CascadesContext) scheduleContext;
+            Stopwatch watch = context.getStopwatch();
+            // when cost time > 10s. we will stop this job.
+            if (watch.elapsed(TimeUnit.MILLISECONDS) > 10000) {
+                throw new RuntimeException("Nereids cost too much time ( > 10s )");
+            }
```

Comment:
the StopWatch's doc say
```
Note that the overhead of measurement can be more than a microsecond
```
so, if we have more than 1000 jobs, then the overhead will more than 1 second, and it is unacceptable